src/core/stdc/math.d

```diff
@@ -215,65 +215,152 @@ version( CRuntime_DigitalMars )
     }
   }
 }
-else version( CRuntime_Microsoft )
+else version( CRuntime_Microsoft ) // fully supported since MSVCRT 12 (VS 2013) only
 {
+  version( all ) // legacy stuff to be removed in the future
+  {
     enum
     {
-        ///
         _FPCLASS_SNAN = 1,
-        ///
         _FPCLASS_QNAN = 2,
-        ///
         _FPCLASS_NINF = 4,
-        ///
         _FPCLASS_NN   = 8,
-        ///
         _FPCLASS_ND   = 0x10,
-        ///
         _FPCLASS_NZ   = 0x20,
-        ///
         _FPCLASS_PZ   = 0x40,
-        ///
         _FPCLASS_PD   = 0x80,
-        ///
         _FPCLASS_PN   = 0x100,
-        ///
         _FPCLASS_PINF = 0x200,
     }
 
-    ///
+    deprecated("Please use the standard C99 function copysignf() instead.")
     float _copysignf(float x, float s);
-    ///
+
+    deprecated("_chgsignf(x) is a non-standard MS extension. Please consider using -x instead.")
     float _chgsignf(float x);
+
+    version( Win64 ) // not available in 32-bit runtimes
+    {
+        deprecated("Please use the standard C99 function isfinite() instead.")
+        int _finitef(float x);
+
+        //deprecated("Please use the standard C99 function isnan() instead.")
+        int _isnanf(float x);
+
+        deprecated("Please use the standard C99 function fpclassify() instead.")
+        int _fpclassf(float x);
+    }
+
+    deprecated("Please use the standard C99 function copysign() instead.")
+    double _copysign(double x, double s);
+
+    deprecated("_chgsign(x) is a non-standard MS extension. Please consider using -x instead.")
+    double _chgsign(double x);
+
+    deprecated("Please use the standard C99 function isfinite() instead.")
+    int _finite(double x);
+
+    //deprecated("Please use the standard C99 function isnan() instead.")
+    int _isnan(double x);
+
+    deprecated("Please use the standard C99 function fpclassify() instead.")
+    int _fpclass(double x);
+  }
+
+    enum
+    {
+        ///
+        FP_SUBNORMAL = -2,
+        ///
+        FP_NORMAL    = -1,
+        ///
+        FP_ZERO      =  0,
+        ///
+        FP_INFINITE  =  1,
+        ///
+        FP_NAN       =  2,
+    }
+
+    private short _fdclass(float x);
+    private short _dclass(double x);
+
+    private int _fdsign(float x);
+    private int _dsign(double x);
+
+  extern(D)
+  {
+    //int fpclassify(real-floating x);
     ///
-    int _finitef(float x);
+    int fpclassify()(float x)   { return _fdclass(x); }
     ///
-    version(Win64) int _isnanf(float x); // not available in Win32?
+    int fpclassify()(double x)  { return _dclass(x);  }
     ///
-    int _fpclassf(float x);
+    int fpclassify()(real x)
+    {
+        static if (real.sizeof == double.sizeof)
+            return _dclass(cast(double) x);
+        else
+            assert(false); // needs to be implemented
```

Comment:
This will always fail with dmd that has 80-bit reals for Win64 and Win32/COFF. I guess we'll have to implement fpclassify for reals before we can use this with dmd...